common/reconciliation/invariants/concreteExecutionExists.go

```diff
@@ -59,6 +60,11 @@ func (c *concreteExecutionExists) Check(execution interface{}) common.CheckResul
 			InvariantType:   c.InvariantType(),
 		}
 	}
+	var runIDCheckResult *common.CheckResult
+	currentExecution, runIDCheckResult = c.validateCurrentRunID(currentExecution)
+	if runIDCheckResult != nil {
+		return *runIDCheckResult
+	}
 	// by this point the corresponding concrete execution must exist and can be already closed
```

Comment:
Can we remove this comment? I think it's kind of misleading. If we are sure that corresponding concrete execution must exist here, then we don't need the `IsWorkflowExecutionExists` call below and can return `CheckResultTypeHealthy` directly I think?

Reply:
Removed